src/lib/markbind/src/parser.js

```diff
@@ -205,7 +205,25 @@ Parser.prototype._preprocess = function (node, context, config) {
     let fileContent = self._fileCache[actualFilePath]; // cache the file contents to save some I/O
     const { parent, relative } = calculateNewBaseUrls(filePath, config.rootPath, config.baseUrlMap);
     const userDefinedVariables = config.userDefinedVariablesMap[path.resolve(parent, relative)];
-    fileContent = nunjucks.renderString(fileContent, userDefinedVariables);
+
+    // process variables declared within the include
+    const includedVariables = context.includedVariables || {};
+    if (element.children) {
+      element.children.forEach((child) => {
+        if (child.name !== 'span') {
+          return;
+        }
+        if (!child.attribs.id) {
+          // eslint-disable-next-line no-console
+          console.warn(`Missing reference in ${element.attribs[ATTRIB_CWF]}\n`
+                     + `Missing 'id' in variable for ${element.attribs.src} include.`);
```

Comment:
1. The "Missing 'id'" part should actually be in front, in order to match the style of other error messages that we show as well.
2. I realise that the "Missing reference..." should be rephrased (because it is not a missing reference, but rather, a missing variable name), apologies about that.

So it should be something like this:

```diff
diff --git a/src/lib/markbind/src/parser.js b/src/lib/markbind/src/parser.js
index 69c40e5..b71b50f 100644
--- a/src/lib/markbind/src/parser.js
+++ b/src/lib/markbind/src/parser.js
@@ -215,8 +215,8 @@ Parser.prototype._preprocess = function (node, context, config) {
         }
         if (!child.attribs.id) {
           // eslint-disable-next-line no-console
-          console.warn(`Missing reference in ${element.attribs[ATTRIB_CWF]}\n`
-                     + `Missing 'id' in variable for ${element.attribs.src} include.`);
+          console.warn(`Missing 'id' in variable for referenced file: ${element.attribs.src}.\n`
+                      + `Missing variable name in ${element.attribs[ATTRIB_CWF]}\n`);
           return;
         }
         includedVariables[child.attribs.id] = cheerio.html(child.children);
```